tests/semanticdb/expect/Advanced.expect.scala

```diff
@@ -34,9 +36,15 @@ object Test/*<-advanced::Test.*/ {
 
   {
     (???/*->scala::Predef.`???`().*/ : Any/*->scala::Any#*/) match {
-      case e3/*<-local9*/: List/*->scala::package.List#*/[_] =>
-        val e3x/*<-local10*/ = e3/*->local9*/.head/*->scala::collection::IterableOps#head().*/
+      case e3/*<-local12*/: List/*->scala::package.List#*/[_] =>
+        val e3x/*<-local14*/ = e3/*->local12*/.head/*->scala::collection::IterableOps#head().*/
         ()
     }
   }
 }
+
+
+// Curried Type Application
+class HKClass/*<-advanced::HKClass#*/[F/*<-advanced::HKClass#[F]*/ <: [T] =>> [U] =>> (U, T)] {
```

Comment:
We are missing occurrences for `[T] =>> [U] =>> (U, T)]` but probably not part of this PR

Reply:
This is the same issue with TypeTree stuff: https://github.com/lampepfl/dotty/pull/12885#discussion_r673987228